### PR TITLE
internal/website: add Cosmos DB to docstore how-to

### LINKED ISSUE #2562: docs/docstore: how-to needs a section for Cosmos
CosmosDB uses the `mongodocstore` driver, but it should have its own section in the HOWTO explaining how to use it.

### PR DESCRIPTION
**Pull request by shantuo**
Fixes #2562 

I added the Cosmos section under the MongoDB section since it uses the mongodocstore package.

### CODE REVIEW

File: internal/website/content/howto/docstore/_index.md [hunk 1]
```diff
@@ -321,10 +321,24 @@ whose name is the combination of two or more fields.
...
 "gocloud.dev/docstore/mongodocstore.ExampleOpenCollectionWithIDFunc" >}}
 
+[`mongodocstore`]: https://godoc.org/gocloud.dev/docstore/mongodocstore
 [`mongodocstore.Dial`]: https://godoc.org/gocloud.dev/docstore/mongodocstore#Dial
 [`mongodocstore.OpenCollection`]: https://godoc.org/gocloud.dev/docstore/mongodocstore#OpenCollection
 [`mongodocstore.OpenCollectionWithIDFunc`]: https://godoc.org/gocloud.dev/docstore/mongodocstore#OpenCollectionWithIDFunc
 
+#### Azure Cosmos DB {#cosmosdb}
+
+[Azure Cosmos DB][] is compatible with MongoDB API. You can use the
```
Comment (vangent) [inline internal/website/content/howto/docstore/_index.md]: Nit: "the MongoDB API"

File: internal/website/content/howto/docstore/_index.md [hunk 2]
```diff
@@ -321,10 +321,24 @@ whose name is the combination of two or more fields.
...
+[`mongodocstore`][] package to connect to Cosmos DB. You must
```
Comment (vangent) [inline internal/website/content/howto/docstore/_index.md]: Can you separate this into a first section about how to use the URLOpener, and then a subheading for the constructor?

In particular, using the URLOpener, you would set the `MONGO_SERVER_URL` to the CosmosDB connection string (maybe link to https://docs.microsoft.com/en-us/azure/cosmos-db/connect-mongodb-account ?).

Comment (shantuo) [reply internal/website/content/howto/docstore/_index.md]: I move the Azure section out and refer to both URL and constructors sections to MongoDB. Let me know whether this is what you meant.

File: internal/website/content/howto/docstore/_index.md [hunk 3]
```diff
@@ -280,13 +280,38 @@ region as your collection:
...
+
 ### MongoDB {#mongo}
 
-The [`mongodocstore`](https://godoc.org/gocloud.dev/docstore/mongodocstore)
-package supports the popular [MongoDB](https://mongodb.org) document store.
-MongoDB documents are uniquely identified by a field called `_id`. In Docstore,
-you can choose a different name for this field, or provide a function to extract
-the document ID from a document.
+The [`mongodocstore`][] package supports the popular
+[MongoDB](https://mongodb.org) document store. MongoDB documents are uniquely
```
Comment (polinasok) [inline internal/website/content/howto/docstore/_index.md]: this links is set up wrong too, isn't it?

**Pull request merged**